### DISCUSSION TITLE
Kmers and the Assembly Graph:  "GFA used k-mer length (k=55) must match the command line settings (k=21)"

### DISCUSSION THREAD

**Discussion by nickp60** [ablab/spades#1188] [Q&A] [answered]
### Description of bug

I am working on a pipeline of running a metagenomic assembly with metaSPAdes, followed by metaViralSPAdes.  I am attempting to use the assembly graph as input to metaviralspades to save time.  The error I get is stating that the kmer length in the assembly graph does not match the cli settings (I guess the default for metaviralSPAdes is 21.  How do I retrieve the kmer size from the assembly graph, so I can use it to set k properly for metaViralSpades? I can tell it gets parsed [here](https://github.com/ablab/spades/blob/3eed982f35e4021439b3b97012898505543afb14/assembler/src/common/io/graph/gfa_reader.cpp#L38), but I can't make heads nor tails of it.

Thanks in adnvance

### spades.log

[spades.log](https://github.com/ablab/spades/files/12740316/spades.log)

### params.txt

[params.txt](https://github.com/ablab/spades/files/12740314/params.txt)

### SPAdes version

SPAdes-3.15.4

### Operating System

Linux-3.10.0-1160.45.1.el7.x86_64-x86_64-with-glibc2.29

### Python Version

3.8.10

### Method of SPAdes installation

Docker

### No errors reported in spades.log

- [X] Yes

**Comment by asl** [comment, marked as answer]
Hello

You need to use last k-mer length if the input is assembly graph. If you have used default set of k-mers for metaSPAdes (did not alter `-k` option), then it would be 55.

**Reply by nickp60** [reply]
Great, thanks so much!
